apps/blockchain/lib/blockchain/difficulty.ex

```diff
@@ -0,0 +1,94 @@
+defmodule Blockchain.Difficulty do
+  @moduledoc """
+  Module to work with Proof-of-work difficulty. It provides functions to
+  help configure difficulty in a coincoin blockchain
+  """
+
+  alias Blockchain.{Block, ProofOfWork}
+
+  @bit_length 256
+  @not_applicable "n/a"
+
+  def benchmark(hashrate \\ nil) do
+    block = Block.generate_next_block("data")
+    [4, 8, 16, 20, 21]
+    |> benchmark(block, [], hashrate)
+    |> print_data()
+  end
+  def benchmark([], _, acc, _), do: acc
```

Comment:
I think this should be separated from the pervious function. Technically they are not the same function because have different `arity ` but I am not sure about this one 🤔